src/server/components/charts-engine/components/processor/index.ts

```diff
@@ -663,14 +669,23 @@ export class Processor {
                 userConfig = configTabResults.exports as UserConfig;
 
                 hrStart = process.hrtime();
-                jsTabResults = await builder.buildChart({
-                    data,
-                    sources: resolvedSources,
-                    params,
-                    usedParams,
-                    actionParams: normalizedActionParamsOverride,
-                    hooks,
-                });
+
+                if (isControlBuilder(builder)) {
+                    jsTabResults = await builder.buildChart({
+                        data,
+                        params,
+                    } as unknown as Parameters<ControlBuilder['buildChart']>[0]);
+                } else {
+                    jsTabResults = await builder.buildChart({
+                        data,
+                        sources: resolvedSources,
+                        params,
+                        usedParams,
+                        actionParams: normalizedActionParamsOverride,
+                        hooks,
+                    });
+                }
```

Comment:
Why not make some of the arguments optional? Then you won't need a fork for different types